### PR TITLE
Fix: mutate methods but not functions

### PR DESCRIPTION
**Pull request by BackEndTea**
See #97 and #111

### CODE REVIEW

File: tests/Mutator/ReturnValue/AbstractValueToNullReturnValueTest.php [hunk 1]
```diff
@@ -137,26 +171,33 @@ public function test_it_does_not_mutate_when_function_contains_another_function_
     {
```
Comment (theofidry) [inline tests/Mutator/ReturnValue/AbstractValueToNullReturnValueTest.php]: Actually I'm not sure about this. Could we double check that if you have:

```php
class Test
{
    function __invoke($x): void {
        $foo = function ($y): bool {
            return true;
        }

        return $foo($x);
    }
}
```

Then _maybe_ the mutants changing the `return true` to `return false` does work. I think the issue with function is only named functions not anonymous functions like that. If that's the case then even the following should work:

```php
$x = function foo() {
    return true; 
}

// As opposed of:

function foo() {
    // the content of this function (namespaced or not) cannot not be mutated
}
```

Comment (borNfreee) [reply tests/Mutator/ReturnValue/AbstractValueToNullReturnValueTest.php]: Actually, this is not a valid syntax. Anonymous function can't have a name.
```php
$x = function foo() {
    return true; 
}
```

I checked your example in this branch and there is a correct mutant:

```diff
class Test
{
    function __invoke($x): void {
        $foo = function ($y): bool {
-            return true;
+            return false;
        }

        return $foo($x);
    }
}
```

**Comment by borNfreee** [comment]
@theofidry do you still have any concerns about this PR (please see https://github.com/infection/infection/pull/113#discussion_r161058354).

If not, I think it can be merged.

**Comment by borNfreee** [comment]
Could you please rebase before merging?

**Comment by BackEndTea** [comment]
Rebased

**Comment by borNfreee** [comment]
Thank you @BackEndTea

**Comment by theofidry** [comment]
@borNfreee back from holidays :) No it looks ok, my concern was that anonymous functions like those should still be mutated

**Pull request merged**